simulator/libc_argp.h

```diff
@@ -0,0 +1,65 @@
+#include <stdlib.h>
+#include <argp.h>
+
+/* Program documentation. */
+static char doc[] =
+    "FOME Simulator -- https://wiki.fome.tech/";
+
+/* A description of the arguments we accept. */
+static char args_doc[] = "";
+
+/* The options we understand. */
+static struct argp_option options[] = {
+    {"quiet", 'q', 0,         0, "Don't produce verbose output", 0 },
+    {"socketcan-device",
+                'd', "DEVICE",  0, "SocketCAN DEVICE (default: can0) to use", 0 },
+    {"timeout", 't', "SECONDS", 0, "Run for SECONDS and then exit (negative values ignored)", 0 },
+    { 0, 0, 0, 0, 0, 0 }
+};
+
+/* Used by main to communicate with parse_opt. */
+struct arguments
+{
+    int quiet;
+    char * socketcanDevice;
+    int timeout;
+};
+
+/* Parse a single option. */
+static error_t
+parse_opt(int key, char * arg, struct argp_state * state)
+{
+    /* Get the input argument from argp_parse, which we
+       know is a pointer to our arguments structure. */
+    struct arguments * arguments = (struct arguments *)state->input;
+
+    switch (key) {
+        case 'd':
+            arguments->socketcanDevice = arg;
+            break;
+        case 'q':
+            arguments->quiet = 1;
+            break;
+        case 't':
+            arguments->timeout = atoi(arg);
+            break;
+
+        case ARGP_KEY_ARG:
+            /* if (state->arg_num >= 0) */ {
```

Comment:
wanted to retain this format in case we add arguments later, but the comparison results in compile error due to `unsigned int` comparison w/ `0` is always true.

```
libc_argp.h:48:32: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
   48 |             if (state->arg_num >= 0) {
```